lib/views/survey_detail/survey_detail_view.dart

```diff
@@ -24,13 +33,22 @@ class SurveyDetailViewState extends ConsumerState<SurveyDetailView>
         width: MediaQuery.of(context).size.width,
         height: MediaQuery.of(context).size.height,
         child: Consumer(builder: (_, ref, __) {
-          // TODO: - Network image stream
-          return Image(image: Assets.images.nimbleBackground.image().image);
+          final uiModel = ref.watch(surveyDetailUiModelStream).valueOrNull;
+          final imageUrl = uiModel?.imageUrl ?? '';
+          return imageUrl.isEmpty
+              ? Image(image: Assets.images.nimbleBackground.image().image)
+              : FadeInImage.assetNetwork(
+                  placeholder: Assets.images.nimbleBackground.path,
+                  image: imageUrl,
+                  fit: BoxFit.cover,
+                  alignment: Alignment.center,
+                );
         }),
       );
 
   Widget get _mainBody => Consumer(
         builder: (_, ref, __) {
+          final uiModel = ref.watch(surveyDetailUiModelStream).valueOrNull;
```

Comment:
@nmint8m I think the `uiModel` should not be nullable. Or we can check null before returning the widget instead of use checking condition 🤔